src/components/nodes/boundaryEvent/boundaryEvent.vue

```diff
@@ -76,32 +81,87 @@ export default {
       this.shape.position(x - (width / 2), y - (width / 2));
       this.updateCrownPosition();
     },
-    attachBoundaryEventToTask() {
-      const task = this.getTaskUnderShape();
+    attachBoundaryEventToTask(task) {
+      const currentlyAttachedTask = this.shape.getParentCell();
 
-      if (!task) {
-        return;
+      if (currentlyAttachedTask) {
+        currentlyAttachedTask.unembed(this.shape);
       }
 
       task.embed(this.shape);
       this.node.definition.set('attachedToRef', task.component.node.definition);
       this.toggleInterruptingStyle(this.node.definition.cancelActivity);
       this.updateShapePosition(task);
+    },
+    moveBoundaryEventIfOverTask() {
+      const task = this.getTaskUnderShape();
 
-      this.shape.listenTo(this.paper, 'element:pointerup', cellView => {
+      if (!task) {
+        this.shape.position(this.validPosition.x, this.validPosition.y);
+        this.updateCrownPosition();
+        return;
+      }
+
+      if (task === this.shape.getParentCell()) {
+        this.updateShapePosition(task);
```

Comment:
`attachBoundaryEventToTask` includes a call to `updateShapePosition` on line 94. Is it necessary to do it here and then again in the method?